content/chainguard/migration-guides/tutorial/index.md

```diff
@@ -0,0 +1,732 @@
+rather than the production one, you can specify it during docker build:
+
+```bash
+docker build --target dev -t identidock:dev .
+```
+
+Otherwise, you will get the production image only.
+
+There are several more environment variables defined. These prevent the creation of Python bytecode
+and buffering of output. The reasons why you want these are normally set are explained in detail in
+the blog [PYTHONDONTWRITEBYTECODE and PYTHONUNBUFFERED
+Explained](https://blog.mimixtech.com/pythondontwritebytecode-and-pythonunbuffered-explained).
+
+The installation of pip modules has moved to the `requirements.txt` file. The main thinking here is
+that we don't need to update the Dockerfile each time a dependency is updated or changed.
+
+The development server runs on port 5000, while the production server runs on port 9090. We could
+edit this so they both run on the same port, but this approach reduces the chance of accidentally
+running the development server in production. The development server is started directly from the
+entrypoint, so we are no longer dependent on an entrypoint script, simplifying our architecture.
+
+To get a minimal, clean production install, we are using a Python virtual environment
+([venv](https://docs.python.org/3/library/venv.html)) in the development image to isolate all
+dependencies, which are then copied over to the production image. Finally, the production image has
+been changed to use [gunicorn](https://gunicorn.org/) as [uwsgi has entered "maintenance
+mode"](https://github.com/unbit/uwsgi).
+
+Build the final image:
+
+```bash
+docker build -t identidock-cg .
+```
+
+And take a look at it:
+
+```bash
+docker images identidock-cg
+REPOSITORY      TAG       IMAGE ID       CREATED          SIZE
+identidock-cg   latest    30a424aa8a51   15 seconds ago   93.6MB
+```
+
+Run a scan with grype:
+
+
+```bash
+grype docker:identidock-cg
+✔ Vulnerability DB                [updated]
+✔ Loaded image                                                                                                                                                                    identidock-cg:latest
+✔ Parsed image                                                                                                                 sha256:30a424aa8a51548491ea3c152383ce2c16a801859e9e8e252b147162ac0b637e
+✔ Cataloged contents                                                                                                                  40024c9c378c73d52c9e62f32fcf6d0d328d8d14a66a328df8d6fc0278a4a2e8
+├── ✔ Packages                        [46 packages]
+├── ✔ File digests                    [2,020 files]
+├── ✔ File metadata                   [2,020 locations]
+└── ✔ Executables                     [141 executables]
+✔ Scanned for vulnerabilities     [0 vulnerability matches]
+├── by severity: 0 critical, 0 high, 0 medium, 0 low, 0 negligible
+└── by status:   0 fixed, 0 not-fixed, 0 ignored
```

Comment:
I also got vulns here

Reply:
Try again with docker build --pull. I think you have old images which also explains the size issue

Reply:
May change over time, but lgtm atm.